File: jre/java/AmpliApp/src/main/java/org/example/LoggingMiddleware.java
package org.example;

import com.amplitude.*;

import org.json.JSONException;

public class LoggingMiddleware implements Middleware {
    @Override
    public void run(MiddlewarePayload payload, MiddlewareNext next) {
        String eventType = null;
        String eventProperties = null;
        if (payload.event != null) {
            eventType = payload.event.eventType;
            if (payload.event.eventProperties != null) {
                try {
                    eventProperties = payload.event.eventProperties.toString(2);
                } catch (JSONException e) {
                    System.err.println(e.toString());
                }
            }
        }

        String extra = null;
        if (payload.extra != null) {
            extra = payload.extra.toString();
        }

        System.out.printf("[ampli] event=%s, properties=%s, extra=%s%n", eventType, eventProperties, extra);

        next.run(payload);
    }
}

File: jre/kotlin/AmpliApp/src/main/kotlin/LoggingMiddleware.kt
import com.amplitude.*

class LoggingMiddleware : Middleware {
    override fun run(payload: MiddlewarePayload?, next: MiddlewareNext?) {
        val event = payload?.event
        println("[ampli] event=${event?.eventType}, eventProperties=${event?.eventProperties?.toString(2)}, userProperties=${event?.userProperties?.toString(2)}, groupProperties=${event?.groupProperties?.toString(2)}, extra=${payload?.extra}")

        next?.run(payload)
    }
}